### DISCUSSION TITLE
Stuck in installing npm dependencies

### DISCUSSION THREAD

**Discussion by Rowee13** [infinitered/ignite#2359] [Q&A]
I am stuck in installing npm dependencies and there's no indication of what is happening if it's downloading files or what.

Here's a screenshot of my terminal.

![DB04BDA5-6779-43AD-97E3-52AA9AE3D655](https://user-images.githubusercontent.com/75659500/212859157-0f73bbde-63a6-4b75-85f2-f329db7a51c1.jpeg)

Does it really take that long? I have a stable internet connection and it is fast enough to download GB of files in minutes.

**Comment by frankcalise** [comment]
Hey! it shouldn't take that long, try running with the `--debug` switch for extra logging to see if it is hanging on something in particular

**Reply by Rowee13** [reply]
I forced it to stop because ctrl+c won't work. I think all the files are downloaded but the terminal was stuck in that note installing npm dependencies. I already tried to run expo start and it already worked on the ios simulator. Do you think it is good now? Sorry about this, this is my first time using the boilerplate. Saw it on react native docs.

**Reply by frankcalise** [reply]
Yeah if the demo is running fine I wouldn't worry about it (you can run `yarn` or `npm install` again to make sure). Appreciate you trying out Ignite!

If you try to fire up another new project and the same thing happens, try throwing the `--debug` switch on to grab more info and maybe we can help debug your setup so it doesn't stall.

Feel free to hop in our Community Slack too, maybe someone has come across it there as well. https://community.infinite.red/

**Reply by jamonholmgren** [reply]
@frankcalise I wonder if we could have a timer running, and if it takes longer than X minutes we could abort the whole process and tell them "This was taking much longer than expected; please run again with `--debug` and file an issue if it keeps happening." ... or something similar.

**Reply by jamonholmgren** [reply]
cc @joshuayoes

**Comment by Rowee13** [comment]
I tried running it again with --debug and here's the result. It stuck at some point again.

![D15E5575-537E-4792-98D5-C6507E7C6155](https://user-images.githubusercontent.com/75659500/213056315-fda0de9c-dae4-4437-b937-8e88e8b1d335.jpeg)

![71AA9481-F583-4EDF-992E-7495E53F8665](https://user-images.githubusercontent.com/75659500/213056077-b1403e93-df3d-4c64-8660-7cafadde6a83.jpeg)

btw, I've got the latest version of nodeJS 18.13,  do you think it has something to do with it? I've got warnings too when running npx create-expo-app

**Reply by frankcalise** [reply]
Have you built react-native apps before? Curious because it looks like some of your tooling could be off, like that Cocoapods install barking about the password being needed to install it. So it could be something related to that.

https://guides.cocoapods.org/using/getting-started.html

You can run `npx ignite-cli doctor` and paste the output here as well for some details about your tooling

**Reply by Rowee13** [reply]
Oh I get it now. Maybe because I haven't tried other tooling in the boilerplate? I haven't yet built extensively, mainly using expo only. I wanted to try Ignite to explore more. Thanks for the help

**Reply by frankcalise** [reply]
@Rowee13 yes very likely. Expo shields you from that stuff which is the great part about it. Although we incorporate Expo into our boilerplate, we do allow for a less Expo heavy path (which we call Vanilla). Both of these options are included in the one install.

Because of that, it's likely you haven't set up a few additional tools if you've only done Expo before - not to worry, I was this way too!

If you'd like Ignite + Expo and not have to worry about native at all, you can follow this guide I wrote to clean up the native vanilla files and trim things up: https://infinitered.github.io/ignite-cookbook/docs/recipes/PristineExpoProject
